pilot/pkg/model/validation.go

```diff
@@ -43,8 +43,10 @@ const (
 	// a wild-card prefix is an '*', a normal DNS1123 label with a leading '*' or '*-', or a normal DNS1123 label
 	wildcardPrefix string = `\*|(\*|\*-)?(` + dns1123LabelFmt + `)`
 
-	// TODO: there is a stricter regex for the labels from validation.go in k8s
-	qualifiedNameFmt string = "[-A-Za-z0-9_./]*"
+	// Using kubernetes requirement, a valid tag must be an empty string or consist
+	// of alphanumeric characters, '-', '_' or '.', and must start and end with an
+	// alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345'
+	qualifiedNameFmt string = "(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?"
```

Comment:
IN k8s it’s
```
const qnameCharFmt string = "[A-Za-z0-9]"
const qnameExtCharFmt string = "[-A-Za-z0-9_.]"
const qualifiedNameFmt string = "(" + qnameCharFmt + qnameExtCharFmt + "*)?" + qnameCharFmt
```

Not the same as above

Reply:
Yes, the difference is that one allows empty strings and the other does not. Made changes to allow empty strings for label values and not label keys. This is similarly done in Kubernetes with labelValueFmt.